src/NHibernate/Hql/Ast/ANTLR/Util/JoinProcessor.cs

```diff
@@ -168,12 +168,12 @@ private static SqlString ProcessFromFragment(SqlString frag, JoinSequence join)
 
 		private static bool HasDynamicFilterParam(SqlString sqlFragment)
 		{
-			return sqlFragment.IndexOfCaseInsensitive(ParserHelper.HqlVariablePrefix) < 0;
```

Comment:
Actually I don't understand this code. Why "has dynamic filter param" is true when HqlVariablePrefix is missing?